PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -38,24 +40,47 @@ def __generate_seed(mnemonic: str, seed_password: str = '') -> bytes:
 # - dictionary is too big (2048 lines OK, but too long words) like hundreds of MB...
 # - every line has exactly 1 word (no whitespaces)
 
-def __entropy2mnemonic(entropy: bytes) -> str:
-    """Convert entropy to mnemonic phrase using dictionary.
+def __get_dictionary() -> Tuple[List[str], Dict[str, int]]:
+    """Load the dictionary.
     Currently uses 1 default dictionary with English words.
     # TODO Should we support multiple dictionaries for various languages?
+    # TODO implement soma checks
+    :rtype: Tuple[List[str], Dict[str, int]]
+    :return: List and dictionary of words
+
+    """
+    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'english.txt'), 'r') as f:
+        l = f.read().splitlines()
+    d = {l[i]: i for i in range(len(l))}
+    return (l, d)
+
+
+def __entropy2mnemonic(entropy: bytes) -> str:
+    """Convert entropy to mnemonic phrase using dictionary.
     :rtype: str
     :return: Mnemonic phrase
     """
-    pass
+    word_list, _ = __get_dictionary()
+    checksum = hashlib.sha256(entropy).digest()
```

Comment:
Need to check that entropy has the correct length.

Reply:
I believe it should be checked before its passed to this function